tortuosity.py

```diff
@@ -31,37 +43,87 @@ def process(self):
         inflection_indices = []
         # not including the first and last elements, this is the middle point for every 3 stations
         for idx in range(1, len(self.stations) - 1):
-            finder = InflectionPointFinder(self.stations[idx - 1: idx + 2])
-            is_inflection = finder.process()
+            is_inflection = find_inflection(self.stations[idx - 1: idx + 2])
             self.stations[idx].is_inflection = is_inflection
             if is_inflection:
                 inflection_indices.append(idx)
 
-        if len(inflection_indices) == 0:
+        if not inflection_indices:
             print("Strange, no inflection point.")
-        else:
-            inflection_indices = [0] + inflection_indices + [len(self.stations) - 1]
-            self.stations[0].is_inflection = True
-            self.stations[-1].is_inflection = True
-            print("Inflection points indices: ", inflection_indices)
+            return
+
+        # adding the first and last points as inflection points
+        inflection_indices = [0] + inflection_indices + [len(self.stations) - 1]
+        self.stations[0].is_inflection = True
+        self.stations[-1].is_inflection = True
+        print("Inflection points indices: ", inflection_indices)
+
+        for idx in inflection_indices[1:]:
+            calculate_and_set_single_tortuosity(self.stations[idx - 1], self.stations[idx])
+
+        df = pd.DataFrame.from_records([s.to_dict() for s in self.stations])
+
+        df['total_ti'] = 0
+        last_total_ti = 0
+        for idx, row in df.iterrows():
```

Comment:
The variable names within this function are too terse to understand easily. Can you expand them so we know what they are about, and also comment on any calculations, so that it's clear what is happening?